docs/pages/release_notes.md

```diff
@@ -1,6 +1,113 @@
 Release Notes {#releasenotes}
 ==============
 
+v3.5.0
+==============
+
+Major Updates
+-------------
+
+* ArrayFire now supports threaded applications
+  <sup>[1](https://github.com/arrayfire/arrayfire/pull/1706)</sup>
+* Added Canny edge detector
+  <sup>[1](https://github.com/arrayfire/arrayfire/pull/1743)</sup>
+* Added Sparse-Dense arithmetic operations
+  <sup>[1](https://github.com/arrayfire/arrayfire/pull/1696)</sup>
+
+Features
+--------
+
+* ArrayFire Threading
+  * \ref af::array can be read by multiple threads
+  * All ArrayFire functions can be executed concurrently by multiple threads.
+  * Threads can operate on different devices to simplify Muli-device workloads
+* New Canny edge detector function, \ref af::canny().
+  <sup>[1](https://github.com/arrayfire/arrayfire/pull/1743)</sup>
+  * Can automatically calculate high threshold with `AF_CANNY_THRESHOLD_AUTO_OTSU`
+  * Supports both L1 and L2 Norms to calculate gradients
+* New tuned OpenCL BLAS backend,
+  [CLBlast](https://github.com/arrayfire/arrayfire/pull/1727).
+
+Improvements
+------------
+
+* Performance improvements when using \ref af::reorder()
+  <sup>[1](https://github.com/arrayfire/arrayfire/pull/1766)</sup>
+* Improved unified backend performance
+  <sup>[1](https://github.com/arrayfire/arrayfire/pull/1770)</sup>
+* ArrayFire now depends on Forge
+  v1.0. <sup>[1](https://github.com/arrayfire/arrayfire/pull/1800)</sup>
+* Can now specify the FFT plan cache size using the
+  \ref af::setFFTPlanCacheSize() function.
+* Get the number of physical bytes allocated by the memory manager
+  \ref `af_get_allocated_bytes()`
+* \ref af::dot() can now return a scalar value to the host.
+
+Bug Fixes
```

Comment:
Also, we fixed complex multiply in convolve kernels complex multiplication. https://github.com/arrayfire/arrayfire/pull/1816